clients/src/test/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkThreadTest.java

```diff
@@ -360,6 +370,52 @@ void testSendUnsentRequest() {
         assertFalse(networkClient.hasAnyPendingRequests());
     }
 
+    @Test
+    void testInvalidTopicMetadataErrorEvent() {
```

Comment:
i don't think we need this test in the consumer network thread as this is indirectly testing network client delegate.

Reply:
it think it would be really great if you could create an integration test in `PlaintextConsumerTest` that mimics the topic metadata error.